src/services/identity/IIdentity.ts

```diff
@@ -1,104 +1,123 @@
-	VeramoAgent,
-	VerificationOptions,
-} from '../../types/types';
+	BroadCastStatusListOptions, CheckStatusListOptions, CreateStatusListOptions, CredentialRequest,
+	StatusOptions, UpdateStatusListOptions, VeramoAgent, VerificationOptions,
+} from '../../types/types'
 
 export interface IIdentity {
-	agent?: TAgent<any>;
-	privateStore?: AbstractPrivateKeyStore;
-	initAgent(): TAgent<any>;
-	createAgent?(agentId: string): Promise<VeramoAgent>;
-	createKey(type: 'Ed25519' | 'Secp256k1', agentId?: string): Promise<ManagedKeyInfo>;
-	getKey(kid: string, agentId?: string): Promise<ManagedKeyInfo>;
-	createDid(network: string, didDocument: DIDDocument, agentId?: string): Promise<IIdentifier>;
-	updateDid(didDocument: DIDDocument, agentId?: string): Promise<IIdentifier>;
-	deactivateDid(did: string, agentId?: string): Promise<boolean>;
-	listDids(agentId?: string): Promise<string[]>;
-	resolveDid(did: string): Promise<DIDResolutionResult>;
-	getDid(did: string, agentId?: string): Promise<any>;
-	importDid(did: string, privateKeyHex: string, publicKeyHex: string, agentId?: string): Promise<IIdentifier>;
-	createResource(network: string, payload: ResourcePayload, agentId?: string): Promise<any>;
-	createCredential(
-		credential: CredentialPayload,
-		format: CredentialRequest['format'],
-		statusOptions: StatusOptions | null,
-		agentId?: string
-	): Promise<VerifiableCredential>;
-	verifyCredential(
-		credential: VerifiableCredential | string,
-		verificationOptions: VerificationOptions,
-		agentId?: string
-	): Promise<IVerifyResult>;
-	verifyPresentation(
-		presentation: VerifiablePresentation | string,
-		verificationOptions: VerificationOptions,
-		agentId?: string
-	): Promise<IVerifyResult>;
-	createStatusList2021(
-		did: string,
-		resourceOptions: ResourcePayload,
-		statusOptions: CreateStatusListOptions,
-		agentId: string
-	): Promise<CreateStatusList2021Result>;
-	updateStatusList2021(
-		did: string,
-		statusOptions: UpdateStatusListOptions,
-		publish?: boolean,
-		agentId?: string
-	): Promise<BulkRevocationResult | BulkSuspensionResult | BulkUnsuspensionResult>;
-	broadcastStatusList2021(
-		did: string,
-		resourceOptions: ResourcePayload,
-		statusOptions: BroadCastStatusListOptions,
-		agentId?: string
-	): Promise<boolean>;
-	checkStatusList2021(
-		did: string,
-		statusOptions: CheckStatusListOptions,
-		agentId?: string
-	): Promise<StatusCheckResult>;
-	revokeCredentials(
-		credential: VerifiableCredential | VerifiableCredential[],
-		publish: boolean,
-		agentId?: string
-	): Promise<RevocationResult | BulkRevocationResult>;
-	suspendCredentials(
-		credential: VerifiableCredential | VerifiableCredential[],
-		publish: boolean,
-		agentId?: string
-	): Promise<SuspensionResult | BulkSuspensionResult>;
-	reinstateCredentials(
-		credential: VerifiableCredential | VerifiableCredential[],
-		publish: boolean,
-		agentId?: string
-	): Promise<UnsuspensionResult | BulkUnsuspensionResult>;
+	initAgent(): TAgent<any>
```

Comment:
In TypeScript, you cannot directly create a member variable of an interface because interfaces are used to define contracts for the shape of objects and do not include the concept of member variables or instance state.